enhancements/ingress/haproxy-healthcheckinterval.md

```diff
@@ -0,0 +1,291 @@
+the healthchecks could time out before the expected 5 seconds.  To mitigate this, the validation on the healthcheck
+interval value (`healthCheckInterval`) will have a minimum of 5 seconds.
+
+## Design Details
+
+### Open Questions [optional]
+
+Is there ever a reason to allow users to set the healthcheck interval lower than 5 seconds?
+
+### Test Plan
+
+Unit tests will be added to test the propagation of the `healthCheckInterval` setting and its interaction with the router
+annotation `router.openshift.io/haproxy.health.check.interval`.
+
+E2E tests can validate that the environment variable is present and the HAProxy template is properly constructed.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+No new environment variables are added, but a new one is exposed.  Upgrades to this enhancement should not impact any
+existing installations until a `healthCheckInterval` setting is configured.  Existing route annotations for the
+healthcheck interval will still override the global healthcheck interval.
+
+Downgrades from this enhancement will revert to the previous behavior of using either the existing route annotation, or
+the default of 5 seconds for healthcheck interval.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+The purpose of backend healthchecks in HAProxy are to verify that the destinations to which it intends to deliver the
+traffic
+are still alive and reachable.  HAProxy will stop delivering traffic to faulty destinations.  The healthcheck interval
+must be small enough to ensure the unreachable destination is not used for very long after it fails.
+HAProxy healthchecks are implemented as `TCP SYN` probes.  HAProxy sends the probe, and then closes the connection with
+a `RST` as soon as it gets a `SYN,ACK` response from the destination.
+
+The default healthcheck interval is 2 seconds, but OpenShift has increased it to 5 seconds, which represents a decrease
+in the number of healthchecks.  As mentioned earlier, 10 seconds is considered a very long (perhaps overly long)
+healthcheck interval.
+
+Use of the new `healthCheckInterval` in the `tuningOptions` will change the frequency of healthchecks
+that HAProxy performs on its backends.  There are scenarios where this could either improve or compromise the
+performance of HAProxy.  Increasing the healthcheck interval too much can result in increased 500-level HTTP responses,
+due to backend servers that are no longer available, but haven't yet been detected as such.  Decreasing the healthcheck
```

Comment:
I would expect the connection from HAProxy to an unavailable backend server to time out and HAProxy to then try a different backend server, resulting in added latency but usually not an HTTP 5xx response, unless multiple backend servers are down (see <http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#4-retries> and <http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#4-option%20redispatch>).  Is that not the case?